src/main/scala/io/iohk/ethereum/utils/Config.scala

```diff
@@ -403,6 +401,7 @@ object PruningConfig {
     val pruningMode: PruningMode = pruningConfig.getString("mode") match {
       case "basic" => BasicPruning(pruningConfig.getInt("history"))
       case "archive" => ArchivePruning
+      case "fastsync" => FastSyncPruning(pruningConfig.getInt("history"))
```

Comment:
after FastSync finishes its run, it is switching to  RegularSync and there we need snapshots if we want to prune, and if you read this from config we will always be stuck with fast sync pruning even during regular sync.